crates/bevy_ecs/src/system/commands.rs

```diff
@@ -166,6 +166,21 @@ impl<'a> Commands<'a> {
         self.queue.push(SpawnBatch { bundles_iter });
     }
 
+    /// Equivalent to iterating `bundles_iter` and calling [`Self::spawn_bundle`] on each bundle,
+    /// then collecting the entities.
+    /// If the the entities are not needed, prefer [`Self::spawn_batch`].
+    #[must_use = "If the returned entities are not needed, use [`Self::spawn_batch`] instead."]
+    pub fn spawn_batch_reserved<I>(&mut self, bundles_iter: I) -> Vec<Entity>
+    where
+        I: IntoIterator + Send + Sync + 'static,
+        I::Item: Bundle,
+    {
+        bundles_iter
+            .into_iter()
+            .map(|bundle| self.spawn_bundle(bundle).id())
+            .collect()
+    }
```

Comment:
If we want to have this be closer to `Commands::spawn` we could also have this return something like a `EntitiesCommands` 
```rust
struct EntitiesCommands {
    entities: Vec<Entity>,
    commands: &mut Commands,
}
```

Though this may be over-engineering it?

Reply:
Yeah I think thats probably over-engineering. I like your current impl more.